packages/voting/graphql/resolvers/QuestionTypeDocumentResult.js

```diff
@@ -1,10 +1,25 @@
-const { graphql: { resolvers: { queries: { document: getDocument } } } } = require('@orbiting/backend-modules-documents')
+const search = require('@orbiting/backend-modules-search/graphql/resolvers/_queries/search')
+const getFieldList = require('graphql-list-fields')
 
 module.exports = {
-  async document (result, args, context) {
+  async document (result, args, context, info) {
+    if (result && result.document) {
+      return result.document
+    }
     if (!result || !result.path) {
       return null
     }
-    return getDocument(null, { path: result.path }, context)
+    const fields = getFieldList(info, true)
+    const withoutContent = fields.indexOf('content') === -1
+    return search(null, {
+      withoutContent,
+      withoutRelatedDocs: true,
```

Comment:
```suggestion
    const relatedDocsNamespaces = [
      'meta.series',
      'meta.format',
      'meta.dossier',
      'meta.discussion'
    ]
    const withoutRelatedDocs = !fields.find(field => (
      relatedDocsNamespaces
        .find(namespace => field.startsWith(namespace))
    ))
    return search(null, {
      withoutContent,
      withoutRelatedDocs,
```

Reply:
with dataloaders this would not be necessary because we could move the resolve logic to the resolvers! ;)

Reply:
Agree. And it would potentially speed up querying of multiple documents, as requested docs might point to the same format the dataloader would cache the responce to the same query and only do it once, not for every doc as now. Sadly we didn't and don't have time for cool refactoring projects like this.